### PR TITLE
EVG-3080 Create version event and triggers

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: model/trigger/version.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		return t(e, v)
+	}
+}
+
+func versionSelectors(v *version.Version) []event.Selector {
+	return []event.Selector{
+		{
+			Type: "id",
+			Data: v.Id,
+		},
```
Comment (richardsamuels) [inline model/trigger/version.go]: Need one more:
```
		{
			Type: "object",
			Data: "version",
		},
```

File: model/trigger/version.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	selectors := versionSelectors(v)
+
+	data := commonTemplateData{
+		ID:                v.Id,
+		Object:            "version",
+		Project:           v.Identifier,
+		URL:               fmt.Sprintf("%s/version/%s", ui.Url, v.Id),
+		PastTenseStatus:   v.Status,
+		apiModel:          &api,
+		githubState:       message.GithubStateFailure,
```
Comment (richardsamuels) [inline model/trigger/version.go]: So we don't actually use github status updates with versions. So how about:
1. delete this line, and the one below it
2. Delete the block below this; the one that changes githubState
3. Go into model/trigger/payloads, and wrap the block beginning on line 32 with `if len(data.githubDescription) != 0 {...}`

File: model/trigger/version.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	}
+
+	selectors := versionSelectors(v)
+
+	data := commonTemplateData{
+		ID:                v.Id,
+		Object:            "version",
+		Project:           v.Identifier,
+		URL:               fmt.Sprintf("%s/version/%s", ui.Url, v.Id),
+		PastTenseStatus:   v.Status,
```
Comment (richardsamuels) [inline model/trigger/version.go]: evergreen.VersionSucceeded actually has a value of "success" but for grammatical correctness in messages, this value needs to be "succeeded", so you'll need an if block.

File: model/trigger/version.go [hunk 4]
```diff
@@ -76,18 +79,18 @@ func generatorFromVersion(triggerName string, v *version.Version) (*notification
...
-		data.githubState = message.GithubStateSuccess
+		pastTense = "succeeded"
+	}
+
+	data := commonTemplateData{
+		ID:              v.Id,
+		Object:          "version",
+		Project:         v.Identifier,
+		URL:             fmt.Sprintf("%s/version/%s", ui.Url, v.Id),
+		PastTenseStatus: v.Status,
```
Comment (richardsamuels) [inline model/trigger/version.go]: this should be pastTense

File: model/trigger/version.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	api := restModel.APIVersion{}
+	if err := api.BuildFromService(v); err != nil {
+		return nil, errors.Wrap(err, "error building json model")
+	}
+
+	selectors := versionSelectors(v)
+
+	pastTense := v.Status
```
Comment (richardsamuels) [inline model/trigger/version.go]: do you think pastTenseStatus would be better?

File: model/trigger/version.go [hunk 6]
```diff
@@ -0,0 +1,130 @@
+package trigger
+
+import (
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/version"
+	restModel "github.com/evergreen-ci/evergreen/rest/model"
```
Comment (tychoish) [inline model/trigger/version.go]: why is there a dependency on the rest model in the db model package? this seems likely to cause strife in the future.

File: model/event/version_event.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+		ResourceType: ResourceTypeVersion,
+		EventType:    VersionStateChange,
+		Data: &VersionEventData{
+			Status: newStatus,
+		},
+	}
+
+	logger := NewDBEventLogger(AllLogCollection)
+	if err := logger.LogEvent(&event); err != nil {
+		grip.Errorf("Error logging version event: %+v", err)
```
Comment (tychoish) [inline model/event/version_event.go]: use a structured message here.

Comment (bsamek) [reply model/event/version_event.go]: done

File: model/trigger/version.go (hunk 6 above)
Comment (bsamek) [reply model/trigger/version.go]: This is taken from the previous notifications. We should push this as is, and discuss refactoring at some other point.

File: model/event/version_event.go (hunk 7 above)
Comment (richardsamuels) [reply model/event/version_event.go]: @bsamek I've added EVG-3392 to add structure to all logging. Can you change the structure of this to match what i described there:
```
message.WrapError(err, message.Fields{
"resource_type": ..., 
"source": "event-log-fail", 
} 
```

File: model/trigger/version.go (hunk 6 above)
Comment (bsamek) [reply model/trigger/version.go]: I added a note to https://jira.mongodb.org/browse/EVG-3374.

**Pull request merged**